sdk/core/Azure.Core/src/Http/HttpHeader.cs

```diff
@@ -12,7 +12,7 @@ public HttpHeader(string name, string value)
         {
             if (string.IsNullOrEmpty(name))
             {
-                throw new ArgumentException(nameof(name));
+                throw new ArgumentException("Name shouldn't be null or empty", nameof(name));
```

Comment:
Make this `$"{nameof(name)} shouldn't be null or empty"` or at least change the case to match the parameter.